src/Register.jsx

```diff
@@ -0,0 +1,181 @@
+import { useState } from 'react';
+import axios from 'axios';
+import { Container, TextField, Button, Typography, Box, FormControlLabel, Checkbox } from '@mui/material';
+import NavBar from './components/NavBar.jsx';
+import { useNavigate } from 'react-router-dom';
+import Footer from './components/Footer.jsx';
+import theme from './theme.jsx';
+import { ThemeProvider } from '@mui/material/styles';
+
+const Register = () => {
+    const [firstName, setFirstName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const [email, setEmail] = useState('');
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [confirmPassword, setConfirmPassword] = useState('');
+    const [isAdmin, setIsAdmin] = useState(false);
+    const [adminPassword, setAdminPassword] = useState('');
+    const [avatar, setAvatar] = useState(null);
+    const [error, setError] = useState('');
+    const navigate = useNavigate();
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        setError('');
+
+        if (password !== confirmPassword) {
+            setError('Passwords do not match');
+            return;
+        }
+
+        try {
+            let role = 'user';
+
+            if (isAdmin) {
+                if (adminPassword !== 'coolcars') {
```

Comment:
users can inspect the contents of scripts loaded on the front end so it would be better to have this password check on the backend, otherwise anyone could sign up as an admin